system-contract-dapp-playground/src/api/hedera/tokenTransfer-interactions/index.ts

```diff
@@ -311,20 +245,47 @@ export const transferSingleTokenFrom = async (
   // invoking contract methods
   try {
     let transactionResult;
-    if (API === 'FUNGIBLE') {
-      transactionResult = await baseContract.transferFromPublic(
-        hederaTokenAddress,
-        sender,
-        receiver,
-        quantity
-      );
-    } else {
-      transactionResult = await baseContract.transferFromNFTPublic(
-        hederaTokenAddress,
-        sender,
-        receiver,
-        quantity
-      );
+
+    switch (API) {
+      case 'FUNGIBLE':
+        transactionResult = await baseContract.transferTokenPublic(
+          hederaTokenAddress,
+          sender,
+          receiver,
+          quantity,
+          { gasLimit }
+        );
+        break;
+
+      case 'NON_FUNGIBLE':
```

Comment:
nit: for consistency
```suggestion
      case 'NFT':
```